src/validators/rdf.js

```diff
@@ -0,0 +1,105 @@
+import XMLDom from 'xmldom';
+
+import { RDF_OBSOLETE_TAGS, RDF_UNALLOWED_TAGS, RDF_UNALLOWED_IF_LISTED_TAGS,
+  VALIDATION_ERROR, VALIDATION_WARNING } from 'const';
+import { RDFParseError } from 'exceptions';
+import * as messages from 'messages';
+
+
+export default class RDFScanner {
+
+  constructor(contents, filename) {
+    this.contents = contents;
+    this.filename = filename;
+    // I don't think this ever needs to be different, but if it does we can
+    // extract the em namespace using:
+    //
+    // this.namespace = this._xmlDoc.documentElement._nsMap.em;
+    this.namespace = 'http://www.mozilla.org/2004/em-rdf#';
+    this.validatorMessages = [];
+    this._xmlDoc = null;
+  }
+
+  scan() {
+    return new Promise((resolve, reject) => {
+      this.mustNotExist()
```

Comment:
I was thinking each “type” of rule check would be its own method, so we'd just have a bunch of calls in `scan()`:

```js
scan() {
  return new Promise((resolve, reject) => {
    this.mustNotExist()
      .then(() => {
        return this.checkForBadValues();
      })
      .then(() => {
        resolve(this.validatorMessages);
      })
      .catch(reject);
  });
}
```
etc.